### DISCUSSION TITLE
TTL в hostlist

### DISCUSSION THREAD

**Discussion by neduus** [bol-van/zapret#301] [Ideas]
Приветствую!
Можно в хостлист добавить значение TTL для домена/хоста? Ну и соответственно обработку пакетов по этим данным.
Работает замечательно, но не со всем оборудованием (конкретные ТВ приставки). Если нужно поясню.

**Comment by bol-van** [comment]
Нет, к хостлисту стратегии отношения не имеют
А множественных стратегий пока не реализовано
Разделять стратегии можно только разными инстансами
Поэтому можно зафиксировать IP приставки и запустить отдельный инстанс tpws или nfqws для него, перенаправить с фильтром source ip приставки

**Reply by bol-van** [reply]
Но более логично на мой взгляд было бы обьединить стратегии в одну и использовать один инстанс
Если разница только в TTL - надо выбрать TTL максимальный из работающих стратегий
И, возможно, добавить доп ограничитель в виде fooling

**Reply by neduus** [reply]
Столкнулся с тем, что иногда приложение на андроид ТВ приставке запускается при ттл фейка например 13 (домен 1), все, что ниже выдает отсутствие соединения и приложение даже не запускается, а видеопоток (домен 2) в это же время может работать при ttl 4, а 13 для него уже много, т.к. кэш ближе основного домена. Домен 1 всегда далеко, а домен 2 гуляет рядом.

**Reply by bol-van** [reply]
тогда нужно переходить либо на autottl (требует тестирования на множестве доменов, подбора delta и оценки устойчивости)
либо применять  другие ограничители типа md5sig, badseq, datanoack

**Reply by almirus** [reply]
@nediis не могу побороть ANDROID TV и стандартный клиент youtube, пробовал разные конфиги, когда было замедление отлично справлялось, а сейчас после включения zapret - приложение на android tv сразу выдает нет сети, при то что остальные устройства в дом сети отлично работают. покажи свой конфиг

**Reply by neduus** [reply]
я не использовал установщик, просто взял бинарник nfqws и на RPI3 запустил в следующей конфигурации, приставки гоняю через RPI3, на самом деле все сложнее, но по существу вот так
если возникает проблема как у меня была можно попробовать ttl увеличивать до тех пор пока не запустится приложение или из листа можно просто удалить основной домен и тогда запускаться будет, но подольше, а после запуска уже будет работать нормально, я вроде как решил эту проблему (она вдруг исчезла -)), но для этого нужно чтобы роутер был микротиком )
да и клиента желательно останавливать перед новой попыткой

nfqws \
--hostlist=y.txt \
--qnum=537 \
--dpi-desync=fake,disorder2 \
--dpi-desync-ttl=4 \
--user=xxxxx \
--daemon --pidfile=/var/run/nfqws.pid
